cli/CommandExecutor.cpp

```diff
@@ -0,0 +1,185 @@
+    const PtrVector<ParseString> *arguments,
+    const CatalogDatabase &catalog_database, FILE *out) {
+  const ParseString &table_name = arguments->front();
+  const std::string &table_name_val = table_name.value();
+  const CatalogRelation *relation =
+      catalog_database.getRelationByName(table_name_val);
+  if (relation == nullptr) {
+     THROW_SQL_ERROR_AT(&(arguments->front())) << " Unrecognized relation "  <<table_name_val;
+  }
+  vector<int> column_widths;
+  int max_attr_column_width = C::kInitMaxColumnWidth;
+  int max_type_column_width = C::kInitMaxColumnWidth;
+
+  for (const CatalogAttribute &attr : *relation) {
+    // Printed column needs to be wide enough to print:
+    //   1. The attribute name (in the printed "header").
+    //   2. Any value of the attribute's Type.
+    int attr_column_width = attr.getDisplayName().length();
+    int type_column_width = attr.getType().getName().length();
+    max_attr_column_width = std::max(max_attr_column_width, attr_column_width);
+    max_type_column_width = std::max(max_type_column_width, type_column_width);
+  }
+  // Add room for one extra character to allow spacing between the column ending and the vertical bar
+  column_widths.push_back(max_attr_column_width+1);
+  column_widths.push_back(max_type_column_width+1);
+
+  fprintf(out, "%*s \"%s\"\n", C::kInitMaxColumnWidth, "Table", table_name_val.c_str());
+  fprintf(out, "%-*s |", max_attr_column_width+1, " Column");
+  fprintf(out, "%-*s\n", max_type_column_width+1, " Type");
+  PrintToScreen::printHBar(column_widths, out);
+  for (const CatalogAttribute &attr : *relation) {
+    fprintf(out, " %-*s |", max_attr_column_width,
+            attr.getDisplayName().c_str());
+    fprintf(out, " %-*s\n", max_type_column_width,
+            attr.getType().getName().c_str());
+  }
+  // TODO(rogers): Add handlers for partitioning information.
+  if (relation->hasIndexScheme()) {
+    fprintf(out, "%*s\n", C::kInitMaxColumnWidth+2, " Indexes");
+    const quickstep::IndexScheme &index_scheme = relation->getIndexScheme();
+    for (auto index_it = index_scheme.begin(); index_it != index_scheme.end();
+         ++index_it) {
+      fprintf(out, "  \"%-*s\" %s", static_cast<int>(index_it->first.length()),
+              index_it->first.c_str(),
+              index_it->second.IndexSubBlockType_Name(
+                  index_it->second.sub_block_type()).c_str());
+      fputc(' ', out);
+      fputc('(', out);
+      fprintf(out, "%s", relation->getAttributeById(index_it->second.indexed_attribute_ids(0))
+                             ->getDisplayName().c_str());
+      for (std::size_t i = 1; i < static_cast<std::size_t>(index_it->second.indexed_attribute_ids_size()); ++i) {
+        const char *attribute_display_name = relation->getAttributeById(
+                                                 index_it->second.indexed_attribute_ids(i))
+                                                 ->getDisplayName().c_str();
```

Comment:
Indentation issue: either add four more whitespace here, or refactor the whole statement as following:

```
  const char * attribute_display_name =
      relation->getAttributeById(index_it->second.indexed_attribute_ids(i))
          ->getDisplayName().c_str();
```